fabric/network.py

```diff
@@ -305,17 +306,29 @@ def normalize(host_string, omit_port=False):
     # takes precedence.
     if user == env.local_user and 'user' in conf:
         user = conf['user']
-    if port == env.default_port and 'port' in conf:
-        port = conf['port']
+
     # Also override host if needed
     if 'hostname' in conf:
         host = conf['hostname']
     # Merge explicit user/port values with the env/ssh_config derived ones
     # (Host is already done at this point.)
     user = r['user'] or user
-    port = r['port'] or port
+
     if omit_port:
         return user, host
+
+    # determine port from ssh config if enabled
+    ssh_config_port = None
+    if env.use_ssh_config:
+        ssh_config_port = conf.get('port', None)
+
+    # port priorities (from highest to lowest)
```

Comment:
This sort of thing seems to me as something that should be documented somewhere. Maybe something like this should go into the docstring here?